### DISCUSSION TITLE
Question about how to start creating template similarly to helm template

### DISCUSSION THREAD

**Discussion by adsouza75** [stefanprodan/timoni#398] [Q&A]
There is flag enabled in our helm template if it true it will install that particular  workload . for if in the value file deployment is enabled it will install deployment. basically we have common helm templates which is used by all the application. can you do same in timoni also , do we need this packages timoniv1 "timoni.sh/core/v1alpha1" , corev1 "k8s.io/api/core/v1" , appsv1 "k8s.io/api/apps/v1"  part of the template

**Comment by nalum** [comment]
Do you mean you want to setup a common deployment spec that is used like a template by other files?

I have a possible example you can see here we define the deployment:

https://github.com/Nalum/cert-manager-module/blob/main/templates/deployment.cue#L12-L99

This is then used further down in the file where we override specific values:

https://github.com/Nalum/cert-manager-module/blob/main/templates/deployment.cue#L101-L115

The `#ControllerDeploymenSpec` comes from this file:

https://github.com/Nalum/cert-manager-module/blob/main/templates/deploymentSpecController.cue#L11

**Reply by adsouza75** [reply]
Looking something like generic template  which will be stored in registry . then there apps use that templates to deploy app using value file this app as company specific. Is there any example for it

**Comment by adsouza75** [comment]
here is sample template that I created for CUE , I'm trying to understand how it will work with timoni . what I am think I will  have set of template deployment , services and HPA.  and when I run pipeline , in the values will provide name of Deployment and other information for example , not sure how ti will work in timoni. Can I connect somebody on teams chat to understand in more , or if you have examples to see , tryin to get head around it

values: {
	name: testdeploy
	metadata: {
		namespace: "test"
		labels: {
		}
	}

	spec: {
		template: {
			metadata: {
			}
			spec: {
				containers: [{

					ports: [{
						containerPort: 443
					}]
					env: [{
						name:  "ASPNETCORE_ENVIRONMENT"
						value: "Development"
					}]
					terminationMessagePath: "/dev/termination-log"
				}, {
				    image: "testings"
     				ports: [{
						containerPort: 443
					}]
					env: [{
						name:  "ASPNETCORE_ENVIRONMENT"
						value: "Development"
					}]
					terminationMessagePath: "/dev/termination-log"
				}]
			}
		}
		revisionHistoryLimit:    10
		progressDeadlineSeconds: 600
	}

}


package kube

deployment: [ID=_]: {
	kind:       "Deployment"
	apiVersion: "apps/v1"
	metadata: {
		name:              ID
		namespace:         string|*"default"
		labels: {
			app:                            ID
			"azure.workload.identity/use":  bool|*true
		}
		annotations: {}
	}

	spec: {
		replicas: int|*1
		selector: matchLabels: app: ID
		template: {
			metadata: {
				labels: {
					app:                           ID
					"azure.workload.identity/use": "true"
				}
			}
			spec: {
				containers: [{
					name:  ID
					image: "algcdhcrmuseamulti.azurecr.io/alegeus.cdh.outboxpublisher:1.0.117-developmajor"

					ports: [...{
						containerPort: int|*80
						protocol:      string|*"TCP"|"UDP"
					}]
					env: [...{
						name:  string
						value: string
					}]
					resources: {
						limits: {
							cpu:    string|*"200m"
							memory: string|*"500Mi"
						}
						requests: {
							cpu:    string|*"100m"
							memory: "string|*300Mi"
						}
					}
					terminationMessagePath?:   string|*"/dev/termination-log"
					terminationMessagePolicy: string|*"File"
					imagePullPolicy:          string|*"IfNotPresent"
				}]
				restartPolicy:                 string|*"Always"
				terminationGracePeriodSeconds: int|*30
				dnsPolicy:                     string|*"ClusterFirst"
				serviceAccountName:            string|*"workload-identity"
				serviceAccount:                string|*"workload-identity"
				securityContext: {}
				schedulerName: string|*"default-scheduler"
			}
		}
		strategy: {
			type: string|*"RollingUpdate"
			rollingUpdate: {
				maxUnavailable: string|*"25%"
				maxSurge:       strin|*"25%"
			}
		}
	}

}

**Comment by adsouza75** [comment]
I was able to get through Now I think I understand how it works
